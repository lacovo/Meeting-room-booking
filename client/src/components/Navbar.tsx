import * as React from 'react';
import { AppBar, Toolbar, Box, Button, Typography } from '@material-ui/core';
import { useHistory } from 'react-router-dom';

export default function Navbar(props: any) {
  const history = useHistory();
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" component="div">
            {props.title}
          </Typography>
          {props.title === 'Booking Table' ? (
            <Button
              className="header-button"
              color="inherit"
              onClick={() => history.push('/roomdetail/0')}
            >
              Create New Booking
            </Button>
          ) : (
            <Button
              className="header-button"
              color="inherit"
              onClick={() => history.push('/')}
            >
              return
            </Button>
          )}
        </Toolbar>
      </AppBar>
    </Box>
  );
}
