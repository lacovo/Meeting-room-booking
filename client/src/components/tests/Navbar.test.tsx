import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Navbar from '../Navbar';
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe('<Navbar />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<Navbar />);
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });
  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<Navbar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
