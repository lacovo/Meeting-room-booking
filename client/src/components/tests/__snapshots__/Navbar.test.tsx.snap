// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> Should render and match the snapshot 1`] = `
<Styled(MuiBox)
  sx={
    Object {
      "flexGrow": 1,
    }
  }
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        component="div"
        variant="h6"
      />
      <WithStyles(ForwardRef(Button))
        className="header-button"
        color="inherit"
        onClick={[Function]}
      >
        return
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</Styled(MuiBox)>
`;
